ntp-daemon/src/system.rs

```diff
@@ -377,137 +365,72 @@ impl<C: NtpClock, T: Wait> System<C, T> {
         }
     }
 
-    fn handle_peer_demobilize(&mut self, index: PeerIndex) {
+    async fn handle_peer_demobilize(&mut self, index: PeerIndex) {
         self.controller.peer_remove(index);
-        self.peers.remove(&index);
+        let state = self.peers.remove(&index).unwrap();
+
+        // Restart the peer reusing its configuration.
+        let spawner_id = state.spawner_id;
+        let peer_id = state.peer_id;
+        let opt_spawner = self.spawners.iter().find(|s| s.id == spawner_id);
+        if let Some(spawner) = opt_spawner {
+            spawner
+                .notify_tx
+                .send(SystemEvent::peer_removed(
+                    peer_id,
+                    PeerRemovalReason::Demobilized,
+                ))
+                .await
+                .expect("Could not notify spawner");
+        }
```

Comment:
has some duplication with the `handle_peer_network_issue`. is there an easy way to combine their logic?